homeassistant/components/nobo_hub/config_flow.py

```diff
@@ -0,0 +1,219 @@
+"""Config flow for Nobø Ecohub integration."""
+from __future__ import annotations
+
+import logging
+import socket
+from typing import Any
+
+from pynobo import nobo
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_COMMAND_OFF, CONF_COMMAND_ON, CONF_IP_ADDRESS
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    CONF_OVERRIDE_TYPE,
+    CONF_OVERRIDE_TYPE_CONSTANT,
+    CONF_OVERRIDE_TYPE_NOW,
+    CONF_SERIAL,
+    CONF_WEEK_PROFILE_NONE,
+    DOMAIN,
+)
+
+DATA_NOBO_HUB_IMPL = "nobo_hub_flow_implementation"
+DEVICE_INPUT = "device_input"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Nobø Ecohub."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the config flow."""
+        self.discovered_hubs = None
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if self.discovered_hubs is None:
+            self.discovered_hubs = await nobo.async_discover_hubs(loop=self.hass.loop)
+
+        errors = {}
+        if user_input is not None:
+            try:
+                serial = user_input.get(CONF_SERIAL)
+                ip_address = user_input.get(CONF_IP_ADDRESS)
+                await self.async_set_unique_id(serial)
+                self._abort_if_unique_id_configured()
+                name = await self._test_connection(serial, ip_address)
+                return self.async_create_entry(title=name, data=user_input)
+            except InvalidSerial:
+                errors["base"] = "invalid_serial"
+            except InvalidIP:
+                errors["base"] = "invalid_ip"
+            except DeviceNotFound:
+                errors["base"] = "device_not_found"
+            except CannotConnect:
+                errors["base"] = "cannot_connect"
+
+        default_suggestion = self._prefill_identifier()
+        return self.async_show_form(
+            step_id="user",
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_SERIAL, default=default_suggestion): str,
```

Comment:
Does the _prefill_identifier overwrite the user typed data from the last step when an error is shown or is that preserved?

Reply:
Good catch - user input is not preserved on error. I'll look into this.

Reply:
Fixed so user input is preserved on errors.